Added Footer inc. Twitter Link

adamrusted opened the pull request:
As per title - included footer as discussed on Slack.
Happy to tweak as necessary - not 100% with `Pug` files.

Commit:
Added Footer inc. Twitter Link

Commit:
Updated Footer Role

ericcornelissen commented on index.pug:
```diff
@@ -112,3 +112,10 @@ html
+        p A <a href="https://github.com/simple-icons/simple-icons/blob/develop/LICENSE.md">CC0</a> project maintained by the <a href="https://github.com/simple-icons/simple-icons/graphs/contributors">Simple Icons Contributors</a>.
+        p Use <a href="https://github.com/simple-icons/simple-icons">GitHub</a> for requests, corrections and contributions.
+      div
+        a(class="footer-share" target="_blank" href="https://twitter.com/intent/tweet?url=https://simpleicons.org&text=Simple%20Icons%3A%20free%20SVG%20icons%20for%20popular%20brands.&via=bathytype") <svg class="footer-share--icon" role="img" viewBox="0 0 24 24" xmlns="http://www.w3.org/2000/svg"><title>Twitter icon</title><path d="M23.953 4.57a10 10 0 01-2.825.775 4.958 4.958 0 002.163-2.723c-.9 ...
```
Haven't looked at what this will look like before/after, but I think this can be made a bit more readable by utilizing some Pug-language features.

Regarding the SVG, perhaps we can use an `<img>` or `<object>` here instead?

```suggestion
        a(
          class="footer-share"
          rel="noopener"
          target="_blank"
          href="https://twitter.com/intent/tweet?url=https://simpleicons.org&text=Simple%20Icons%3A%20free%20SVG%20icons%20for%20popular%20brands.&via=bathytype"
        )
          svg(class="footer-share--icon" role="img" viewBox="0 0 24 24" xmlns="http://www.w3.org/2000/svg")
            title Twitter logo
            path(d="M23.953 4.57a10 10 0 01-2.825.775 4.958 4.958 0 002.163-2.723c-.951.555-2.005.959-3.127 1.184a4.92 4.92 0 00-8.384 4.482C7.69 8.095 4.067 6.13 1.64 3.162a4.822 4.822 0 00-.666 2.475c0 1.71.87 3.213 2.188 4.096a4.904 4.904 0 01-2.228-.616v.06a4.923 4.923 0 003.946 4.827 4.996 4.996 0 01-2.212.085 4.936 4.936 0 004.604 3.417 9.867 9.867 0 01-6.102 2.105c-.39 0-.779-.023-1.17-.067a13.995 13.995 0 007.557 2.209c9.053 0 13.998-7.496 13.998-13.985 0-.21 0-.42-.015-.63A9.935 9.935 0 0024 4.59z")
          span Share this on Twitter!
```

adamrusted replied:
Happy to go with `img` or `object` if they're better? I was wondering if there was a way we could use the SVG from the simple-icons module here, just in case Twitter *ever* update their logo?

ericcornelissen commented on index.pug:
```diff
@@ -112,3 +112,10 @@ html
+    footer(class="footer" role="contentinfo")
+      div
+        p A <a href="https://github.com/simple-icons/simple-icons/blob/develop/LICENSE.md">CC0</a> project maintained by the <a href="https://github.com/simple-icons/simple-icons/graphs/contributors">Simple Icons Contributors</a>.
+        p Use <a href="https://github.com/simple-icons/simple-icons">GitHub</a> for requests, corrections and contributions.
```
Minor, but preferably these links use `rel="noopener"`

```suggestion
        p A <a href="https://github.com/simple-icons/simple-icons/blob/develop/LICENSE.md" rel="noopener">CC0</a> project maintained by the <a href="https://github.com/simple-icons/simple-icons/graphs/contributors" rel="noopener">Simple Icons Contributors</a>.
        p Use <a href="https://github.com/simple-icons/simple-icons" rel="noopener">GitHub</a> for requests, corrections and contributions.
```

ericcornelissen commented on stylesheet.css:
```diff
@@ -464,6 +464,53 @@ body.order-by-relevance .grid-item {
+/* Footer */
+.footer {
+  align-items: center;
+  background: #EEE;
```
I'd personally go with a footer that is just part of the background, like the header, any particular reason you went with this?

adamrusted replied:
No reason particularly, just mimicking the existing site in that regard.

ericcornelissen reviewed (changes requested):
Haven't seen the `<footer>` yet but here are some minor requests for changes up front, looks promising 👍

Commit:
Apply suggestions from code review

Co-authored-by: Eric Cornelissen <ericornelissen@gmail.com>

Commit:
Update stylesheet.css

ericcornelissen committed:
Mobile and a11y improvements

A couple of minor improvements to the footer for mobile/small screens as
well as improvements to the a11y of the footer.

ericcornelissen committed:
Merge branch 'master' into Footer

ericcornelissen reviewed (approved):
Looking good, I took the liberty of fixing some problems mostly related to the mobile layout and a11y, but really didn't change anything major. Before I merge this in, maybe you can have one last look? :slightly_smiling_face:
